onelogin/get.go

```diff
@@ -107,30 +109,80 @@ func Get(app, provider string) (*awsprovider.Credentials, error) {
 		fmt.Scanln(&selection)
 
 		deviceID = fmt.Sprintf("%v", devices[selection-1].DeviceId)
+		deviceType = devices[selection-1].DeviceType
+
 	} else {
 		deviceID = fmt.Sprintf("%v", devices[0].DeviceId)
+		deviceType = devices[0].DeviceType
 	}
 
-	fmt.Print("Please enter the OTP from your MFA device: ")
-	var otp string
-	fmt.Scanln(&otp)
+	var samlAssertion string
+
+	var pushSupported = false
+
+	if deviceType == "OneLogin Protect" {
+		// push is supported by the selected OTP device
+		// if other are supporting push they should be added here.
+		pushSupported = true
+		s.Start()
+		pMfa := VerifyFactorParams{
+			AppId:       a.ID,
+			DeviceId:    deviceID,
+			StateToken:  st,
+			OtpToken:    "",
+			DoNotNotify: false,
+		}
 
-	// Verify MFA
-	pMfa := VerifyFactorParams{
-		AppId:      a.ID,
-		DeviceId:   deviceID,
-		StateToken: st,
-		OtpToken:   otp,
-	}
+		rMfa, err := c.VerifyFactor(token, &pMfa)
+		if err != nil {
+			return nil, err
+		}
+		pMfa.DoNotNotify = true
+		timeout := 30
+		interval := 4
+		fmt.Printf(" %v\r", rMfa.Status.Message)
+		for rMfa.Status.Type == "pending" && timeout > 0 {
+			time.Sleep(time.Duration(interval) * time.Second)
+			rMfa, err = c.VerifyFactor(token, &pMfa)
+			if err != nil {
+				return nil, err
+			}
+
+			fmt.Printf(" %v\r", rMfa.Status.Message)
+			timeout -= interval
+		}
+		if rMfa.Status.Type == "pending" {
+			s.Stop()
+			fmt.Println("Push timed out, please enter OTP token.")
```

Comment:
This message is a bit redundant with the next one. I would rephrase to:

`MFA verification failed - falling back to manual OTP input`

Reply:
I slightly reworded that to `MFA verification timed out - falling back to manual OTP input`

Reply:
But that's the thing - this is also printed when the user hits "reject", so this is why I wrote "failed" and not "timed out"...